src/extension-consumer/src/consume/mod.rs

```diff
@@ -74,6 +92,160 @@ impl ConsumeLogOpt {
             version: env!("CARGO_PKG_VERSION").into(),
         }
     }
+
+    /// Consume records according to the given configuration.
+    pub async fn consume_records(&self, consumer: PartitionConsumer) -> Result<(), ConsumerError> {
+        debug!("starting fetch loop: {:#?}", self);
+        self.init_ctrlc()?;
+        let offset = self.calculate_offset()?;
+
+        let consume_config = {
+            let mut config = ConsumerConfig::default();
+            if let Some(max_bytes) = self.max_bytes {
+                config = config.with_max_bytes(max_bytes);
+            }
+            config
+        };
+
+        if self.disable_continuous {
+            self.consume_records_batch(&consumer, offset, consume_config)
+                .await?;
+        } else {
+            self.consume_records_stream(&consumer, offset, consume_config)
+                .await?;
+        }
+
+        Ok(())
+    }
+
+    /// Consume records in a single batch, then exit
+    async fn consume_records_batch(
+        &self,
+        consumer: &PartitionConsumer,
+        offset: Offset,
+        config: ConsumerConfig,
+    ) -> Result<(), ConsumerError> {
+        let response = consumer.fetch_with_config(offset, config).await?;
+
+        debug!(
+            "got a single response: LSO: {} batches: {}",
+            response.log_start_offset,
+            response.records.batches.len(),
+        );
+
+        for batch in response.records.batches.iter() {
+            for record in batch.records().iter() {
+                let key = record.key.as_ref().map(|it| it.as_ref());
+                self.print_record(key, record.value.as_ref())?;
+            }
+        }
+        Ok(())
+    }
+
+    /// Consume records as a stream, waiting for new records to arrive
+    async fn consume_records_stream(
+        &self,
+        consumer: &PartitionConsumer,
+        offset: Offset,
+        config: ConsumerConfig,
+    ) -> Result<(), ConsumerError> {
+        let mut stream = consumer.stream_with_config(offset, config).await?;
+
+        while let Some(result) = stream.next().await {
+            let result: Result<Record, _> = result;
+            let record = match result {
+                Ok(record) => record,
+                Err(FluvioError::ApiError(ApiError::Code(code, _))) => {
+                    println!(
```

Comment:
this should be output using `eprintln`.
topic and partition information is redundant since it is already specified by the user and repeat on every record.

Reply:
This is directly pulled from old implementation https://github.com/infinyon/fluvio/pull/830/files#diff-fba8aea514e4ee4d3a739e2ccb439a3f43a63f35ab42df364a786a8253b11c95L103-L110

I can change this to eprintln and just print error code sentence 👍